New component called Segmented Progress Bar added in the scorer ui kit

fti-rchaniya opened the pull request:
### Description

This PR consist a new feature added in the scorer ui kit, below is the description:
**Segmented Progress Bar**
- It is a new feature
- For visualizing the progression of the provided data
- Depending upon the data, we have different status like completed, failed, processing, partially completed, so according to these different status we have created a component which will show these status in the segmented progress bar which will be easy to visualize
- Also a tooltip is added to every segment for briefing about the segment with its information.
- There are 4 props (**statusSegments, statusText, width, height**) and none of the props are mandatory, all the props are optional
- Height and width are use to resize progress bar
- Status text (Completed, Failed, Processing, Partially Completed) anything coming from the backend representing the status
- Status segments is an array of object consisting status with its information 

 ### Considerations on Implementation
 This new component is added in the group of **MISC**

 ### Reviewing/Testing steps
https://future-standard.github.io/scorer-ui-kit/storybook/?path=/story/misc--segmented-progress

Here are some screenshots of the result.
![image](https://user-images.githubusercontent.com/118800413/223027592-c38b6014-0034-4c26-b8c4-f5ee502e1150.png)
![image](https://user-images.githubusercontent.com/118800413/223027756-f7ece70f-3bfc-4326-8bcd-03777d673e4e.png)
![image](https://user-images.githubusercontent.com/118800413/223028051-1c5f779b-aa8b-4696-ba85-d771768b1980.png)

fti-skumar commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,112 @@
+  box-sizing: border-box;
+  flex: 1;
+  height: inherit;
+  border: 1px solid #DDD;
```
Don't know why but when this was added the rounded corners are gone, also I think rounded corners should be there on both sides, you might need to define border-radius

![image](https://user-images.githubusercontent.com/95204740/223092078-784e6866-bea9-4a38-8c82-ddad267cedd7.png)

fti-rchaniya replied:
Done

fti-skumar reviewed (changes requested):
Few comments please check,
And,
Below 3 comments are for user experience and I think @fti-bcarpenter sir can suggest on them

1. I think we should change position of Tooltip to bottom because its over the status text

![image](https://user-images.githubusercontent.com/95204740/223089320-788db440-4e24-4052-8694-69da82af59f9.png)

2.  And if we can add a comment-type tooltip like this it will look amazing

![image](https://user-images.githubusercontent.com/95204740/223090732-48c61105-3132-4201-ba6f-3ab534c3bad4.png)

3. There should be a gap between status text and bar maybe 3-5px

![image](https://user-images.githubusercontent.com/95204740/223089659-0f0706c3-1ff2-4597-9906-0480e54165c7.png)

@fti-bcarpenter sir, please suggest on these.

fti-rchaniya commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,111 @@
+          return (
+            item.status === 'completed' || item.status === 'failed' ? <CompletedFailedBar status={item.status}><TooltipText>{item.displayText}</TooltipText></CompletedFailedBar> :
+              item.status === 'processing' ? <ProcessingBar><TooltipText>{item.displayText}</TooltipText></ProcessingBar> : <EmptyBar><TooltipText>{item.displayText}</TooltipText></EmptyBar>
+          );
```
Done

fti-rchaniya commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,111 @@
+  width: ${({ width }) => width ? width : '120px'};
+`;
+
+const TooltipText = styled.span`
```
As there was no any specific tooltip customization is given, hence I replaced tooltip css with the html title attribute.
Done

fti-rchaniya commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,111 @@
+  background-color: #f8f9fa;
+  outline: none !important;
+  position: absolute;
+  top: -750%;
```
As, I have replaced tooltip css with the html title attribute, this css have been removed
Done

fti-rchaniya commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,111 @@
+  box-sizing: border-box;
+  flex: 1;
+  height: inherit;
+  border: ${({status})=> status === 'completed' ? '1px solid #4DB5FF': '1px solid #EF7878'};
```
Done

fti-rchaniya commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,111 @@
+    #4DB5FF 10px
+  );
+  background-size: 400px 400px;
+  animation: barberpole 20s linear infinite;
```
Done

fti-rchaniya commented on storybook/src/stories/Misc/SegmentedProgressBar.stories.tsx:
```diff
@@ -0,0 +1,25 @@
+    const width = text("width", '');
+    const height = text("height", '');
+  
+  return <Container><SegmentedProgressBar {...{statusText, statusSegments, width, height}}></SegmentedProgressBar></Container>;
```
Done

atomworks reviewed (changes requested):
Please check and fix any lint issues.

isacoder commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,126 @@
+const FailedBar = styled(Bar)`
+  border: 1px solid #EF7878;
+  background: #EF7878;
+  cursor: pointer;
```
Why is the Failed bar the only one with pointer?
Maybe add it to all the bars or keep them without since the tooltip is not a clickable element, but a hover one.

isacoder commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,126 @@
+  color: #000000;
+  line-height: 1.5;
+  text-align: left;
+  border: 1px solid hsl(195deg 5% 60% / 72%);
```
Could you please update the border to a more common border?
border: 1px solid ${({theme: {colors}}) => colors.divider};

isacoder commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,126 @@
+  visibility: hidden;
+  padding: 5px;
+  font-size: 10px;
+  color: #000000;
```
I think it would be nice a text color similar to what we are using, for this tooltip I think the closer will be the font color of the dropdown options  hsla(0, 0%, 56.1%, 1.000);

color: hsla(0, 0%, 56.1%, 1.000);

isacoder commented on src/Misc/SegmentedProgressBar.tsx:
```diff
@@ -0,0 +1,126 @@
+  font-family: ${({ theme }) => theme.fontFamily.data };
+`;
+
+interface ISegments {
```
Maybe it would be useful to export the ISegments interface so the developers can use it when they create a variable. It's helpful when managing states.

I recommend updating Segments to IBarSegments to be more meaningful when using it outside of this code.

isacoder reviewed (changes requested):
Hi, I have added two requests and some color suggestions.

- The interface ISegments would be useful to have externally, please update
- The cursor probably should not be pointer only in failure, please review

I don't have de design reference for this new feature but I have added some color suggestions to have it closer to other components. If the design is required as it is for a project, please let me know.

fti-rchaniya commented:
> Hi, I have added two requests and some color suggestions.
> 
> * The interface ISegments would be useful to have externally, please update
> * The cursor probably should not be pointer only in failure, please review
> 
> I don't have de design reference for this new feature but I have added some color suggestions to have it closer to other components. If the design is required as it is for a project, please let me know.

Thanks for your detailed review, allow me some time, once I'll be free from my current task I'll fix all your feedbacks provided in the comment.

atomworks commented:
This PR looks to be abandoned, so I am closing it. Please re-open if / when required and you are able to complete the work. Thank you.
